### PR TITLE
[BugFix] Fix when output has duplicate item, order by works on wrong column-ref

### LINKED ISSUE #13341: [Bug] when the output columnRef contains duplicate item, the order by works on wrong ColumnRef
### Steps to reproduce the behavior (Required)
1. CREATE TABLE tt (c1 int, c2 int, c3 int);
2. explain select c1, * from tt order by c2;

### Expected behavior (Required)
order by: <slot 2> 2: v2 ASC

### Real behavior (Required)
order by: <slot 1>1: v1 ASC
截屏2022-11-14 15 38 48

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`

### PR DESCRIPTION
**Pull request by HangyuanLiu**
## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #13341

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

### CODE REVIEW

**Comment by HangyuanLiu** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 31 / 31 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/transformer/QueryTransformer.java|16|16|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/SelectRelation.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/transformer/ExpressionMapping.java|14|14|100.00%|[]|

**Review by murphyatwork — DISMISSED** [review]

**Comment by Seaven** [comment]
add an error case in describe?

**Review by zombee0 — DISMISSED** [review]
LGTM

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 80 / 82 (97.56%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/analysis/TableName.java|4|5|80.00%|[161]|
|:large_blue_circle:|com/starrocks/sql/analyzer/ViewDefBuilder.java|19|20|95.00%|[68]|
|:large_blue_circle:|com/starrocks/sql/analyzer/SelectAnalyzer.java|21|21|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/transformer/QueryTransformer.java|17|17|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/SelectRelation.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/AnalyzeState.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/transformer/ExpressionMapping.java|13|13|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/QueryAnalyzer.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/AST2SQL.java|3|3|100.00%|[]|

**Comment by HangyuanLiu** [comment]
run starrocks_admit_test

**Comment by HangyuanLiu** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 83 / 85 (97.65%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/analysis/TableName.java|4|5|80.00%|[161]|
|:large_blue_circle:|com/starrocks/sql/analyzer/ViewDefBuilder.java|19|20|95.00%|[68]|
|:large_blue_circle:|com/starrocks/sql/analyzer/SelectAnalyzer.java|24|24|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/transformer/QueryTransformer.java|17|17|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/SelectRelation.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/AnalyzeState.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/transformer/ExpressionMapping.java|13|13|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/QueryAnalyzer.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/AST2SQL.java|3|3|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 81 / 83 (97.59%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/analysis/TableName.java|4|5|80.00%|[161]|
|:large_blue_circle:|com/starrocks/sql/analyzer/ViewDefBuilder.java|19|20|95.00%|[68]|
|:large_blue_circle:|com/starrocks/sql/analyzer/SelectAnalyzer.java|21|21|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/transformer/QueryTransformer.java|17|17|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/SelectRelation.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/AnalyzeState.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/transformer/ExpressionMapping.java|13|13|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/AST2SQL.java|3|3|100.00%|[]|

**Comment by HangyuanLiu** [comment]
run starrocks_admit_test

**Comment by HangyuanLiu** [comment]
run starrocks_admit_test

**Comment by HangyuanLiu** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 82 / 84 (97.62%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/analysis/TableName.java|6|7|85.71%|[161]|
|:large_blue_circle:|com/starrocks/sql/analyzer/ViewDefBuilder.java|19|20|95.00%|[68]|
|:large_blue_circle:|com/starrocks/sql/analyzer/SelectAnalyzer.java|17|17|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/transformer/QueryTransformer.java|22|22|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/SelectRelation.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/AnalyzeState.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/transformer/ExpressionMapping.java|9|9|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/AST2SQL.java|3|3|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/transformer/ExpressionMapping.java [hunk 1]
```diff
@@ -101,6 +101,27 @@ public void put(Expr expression, ColumnRefOperator columnRefOperator) {
...
+
+        expressionToColumns.put(expression, columnRefOperator);
+    }
+
+    public void putWithSymbol(Expr expression, String symbol, ColumnRefOperator columnRefOperator) {
+        if (expression instanceof FieldReference) {
+            fieldMappings[((FieldReference) expression).getFieldIndex()] = columnRefOperator;
+        } else if (expression instanceof SlotRef) {
+            SlotRef s = (SlotRef) expression;
+            if (!s.getColumnName().equals(symbol)) {
```
[the remark this answers is not in the document]
Comment (HangyuanLiu) [reply fe/fe-core/src/main/java/com/starrocks/sql/optimizer/transformer/ExpressionMapping.java]: This location is case-insensitive

**Review by murphyatwork — APPROVED** [review]

**Review by murphyatwork — APPROVED** [review]

**Comment by HangyuanLiu** [comment]
run starrocks_admit_test

**Comment by HangyuanLiu** [comment]
run starrocks_admit_test

**Review by Seaven — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.5

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.4

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.3

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.2

**Pull request merged**